docs/en/understand_mmcv/data_transform.md

```diff
@@ -0,0 +1,337 @@
+As mentioned above, the input and output of all data transformations are a dictionary, and according to the \[Convention on Datasets\] (TODO) in OpenMMLab, the basic information of each sample in the dataset is a dictionary. This way, we can connect all data transformation operations end to end and combine them into a data pipeline. This pipeline inputs the information dictionary of the samples in the data set and outputs the information dictionary after a series of processing.
+
+Taking the classification task as an example, we show a typical data pipeline in the figure below. For each sample, the information stored in the data set is a dictionary, as shown on the far left in the figure. After each data transformation operation represented by the blue block, a new field (marked in green) will be added to the data dictionary or update an existing field (marked in orange).
+
+<div align=center>
+<img src="https://user-images.githubusercontent.com/26739999/154197953-bf0b1a16-3f41-4bc7-9e67-b2b9b323d895.png" width="90%"/>
+</div>
+
+The data pipeline is a list of several data transformation configuration dictionaries in the configuration file. Each dataset needs to set the parameter `pipeline` to define the data preparation operations the dataset needs to perform. The configuration of the above data pipeline in the configuration file is as follows:
+
+```python
+pipeline = [
+    dict(type='LoadImageFromFile'),
+    dict(type='Resize', size=256, keep_ratio=True),
+    dict(type='CenterCrop', crop_size=224),
+    dict(type='Normalize', mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375]),
+    dict(type='ClsFormatBundle')
+]
+
+dataset = dict(
+    ...
+    pipeline=pipeline,
+    ...
+)
+```
+
+## Common data transformation classes
+
+The commonly used data transformation classes can be roughly divided into data loading, data preprocessing and enhancement, and data formatting. In MMCV, we provide some commonly used classes as follows:
+
+### Data loading
+
+To support the loading of large-scale datasets, data is usually not loaded when `Dataset` is initialized. Only the corresponding path is loaded. Therefore, it is necessary to load specific data in the data pipeline.
+
+|            Class            |                    Feature                     |
+| :-------------------------: | :--------------------------------------------: |
+| [`LoadImageFromFile`](TODO) |              Load from file path               |
+|  [`LoadAnnotations`](TODO)  | Load and organize the annotations (bbox, etc.) |
+
+### Data preprocessing and enhancement
+
+Data preprocessing and augmentation usually involve transforming the image itself, such as cropping, padding, scaling, etc.
+
+|              Class               |                        Feature                         |
+| :------------------------------: | :----------------------------------------------------: |
+|          [`Pad`](TODO)           |                        Padding                         |
```

Comment:
Can we fill these empty links now?

Reply:
> Can we fill these empty links now?

I checked the mmcv's API pages, and it seems multiple features have already been finished. However, would the internal team take a look at these? I believe this would be more appropriate. @zhouzaida

Reply:
Hi @xin-li-67 , it would be better to update the empty links. They can be found at https://mmcv.readthedocs.io/en/2.x/api/transforms.html.

Reply:
> Hi @xin-li-67 , it would be better to update the empty links. They can be found at https://mmcv.readthedocs.io/en/2.x/api/transforms.html.

I think it's better to use doc anchors instead of links. If @xin-li-67 is not familiar with doc anchors, we can add it directly.

Reply:
> > Hi @xin-li-67 , it would be better to update the empty links. They can be found at https://mmcv.readthedocs.io/en/2.x/api/transforms.html.
> 
> I think it's better to use doc anchors instead of links. If @xin-li-67 is not familiar with doc anchors, we can add it directly.

Thanks for the comment. I'm not quite familiar with doc anchors indeed. Would you and the team please finish them directly? That will be grateful.